packages/java-parser/src/tokens.js

```diff
@@ -122,8 +122,9 @@ createToken({ name: "OctalLiteral", pattern: /0_*[0-7]([0-7_]*[0-7])?[lL]?/ });
 createToken({
   name: "FloatLiteral",
   pattern: MAKE_PATTERN(
-    "{{Digits}}\\.({{Digits}})?({{FloatTypeSuffix}})?|" +
-      "\\.{{Digits}}({{ExponentPart}})?({{FloatTypeSuffix}})?|" +
+    "\\.{{Digits}}({{ExponentPart}})?({{FloatTypeSuffix}})?|" +
```

Comment:
Lets try to refactor this to align strongly with the specifications.
https://docs.oracle.com/javase/specs/jls/se11/html/jls-3.html#jls-FloatingPointLiteral

Examples:
- The Spec has 4 variations but now we have 5.
- In the spec the first variant starts with a digit, but our first variant starts with a "dot".

The stronger we align with the spec's **style** the easier it is to prove
that we are aligned with the spec's actual meaning...